opendatahub-io/notebooks
# Pre-install some extensions on code-server notebook

Related to: https://github.com/opendatahub-io/notebooks/issues/336

## Description
By default, code-server does not come with any installed extensions. This pull request aims to streamline the user experience by pre-installing several useful extensions on behalf of the end user. The extensions that we added are:
- Jupyter (ms-toolsai.jupyter)
- python (ms-python.python)

Test image: `quay.io/rh_ee_atheodor/workbench-images@sha256:ac16959bffb660397247100666729b314eb78a8b0ec387131b878697d3cb1e48`

## How Has This Been Tested?
1. Spin up ODH 
2. Edit the `code-server-notebook` imagestream by adding the following tag, Builds > ImageStreams
```
    - name: 'new'
      annotations:
        opendatahub.io/notebook-python-dependencies: '[{"name":"code-sever","version":"4.11"}]'
        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.9"}]'
        openshift.io/imported-from: quay.io/opendatahub/workbench-images
      from:
        kind: DockerImage
        name: >-
         quay.io/rh_ee_atheodor/workbench-images@sha256:ac16959bffb660397247100666729b314eb78a8b0ec387131b878697d3cb1e48
      importPolicy: {}
      referencePolicy:
        type: Source
```
3. Enter on DSP and trigger a vs code using the `new` tag that you created above.
4. Inspect on the extension tab, and check if there are already installed extensions.
You should see the following:
![image](https://github.com/opendatahub-io/notebooks/assets/42587738/08cbfe3f-882d-4ff0-a354-3638dbd1d326)

5. Or by a terminal should type: `code-server --list-extensions`
You should see the following:
![image](https://github.com/opendatahub-io/notebooks/assets/42587738/a24e503e-4fb6-4663-af42-f76db3a37ed0)

6. To locally test the notebook in air-gap env disconnect your internet connection and trigger by using
`podman run --network=host --name validation-container quay.io/rh_ee_atheodor/workbench-images@sha256:ac16959bffb660397247100666729b314eb78a8b0ec387131b878697d3cb1e48`

## Merge criteria:
